File: benchmarks/dot.js
const N = 100000;
let arrayA = new Array(N);
let arrayB = new Array(N);
let sum = 0;

// Initialize arrays with constants
for (let i = 0; i < N; i++) {
    arrayA[i] = i + 1;
    arrayB[i] = N - i;
}

// Compute the dot product
for (let i = 0; i < N; i++) {
    sum += arrayA[i] * arrayB[i];
}

// Print the result
console.log(sum);

File: benchmarks/dot.py
N = 100000
array_a = [i + 1 for i in range(N)]
array_b = [N - i for i in range(N)]
sum = 0

# Compute the dot product
for i in range(N):
    sum += array_a[i] * array_b[i]

# Print the result
print(sum)
